src/Service/Parsers/ODataASTVisitor.cs

```diff
@@ -37,6 +37,17 @@ public override string Visit(BinaryOperatorNode nodeIn)
             // In order traversal but add parens to maintain order of logical operations
             string left = nodeIn.Left.Accept(this);
             string right = nodeIn.Right.Accept(this);
+
+            if (nodeIn.Left.GetType() == typeof(SingleValuePropertyAccessNode) && nodeIn.Right.GetType() == typeof(ConstantNode) ||
+                nodeIn.Left.GetType() == typeof(ConstantNode) && nodeIn.Right.GetType() == typeof(SingleValuePropertyAccessNode))
```

Comment:
Any reason why this is required now?

Reply:
Because we want to populate the DbType for fields referenced in filter/db policy.